samples/cpp/tutorial_code/core/out_of_focus_deblur_filter/out_of_focus_deblur_filter.cpp

```diff
@@ -0,0 +1,141 @@
+﻿/**
+* @file out_of_focus_deblur_filter.cpp
+* @brief You will learn how to recover an out-of-focus image by Wiener filter
+* @author Karpushin Vladislav, karpushin@ngs.ru, https://github.com/VladKarpushin
+*/
+#include "opencv2/opencv.hpp"
+
+using namespace cv;
+using namespace std;
+
+void help();
+void calcPSF(Mat& outputImg, Size filterSize, int R);
+void fftshift(const Mat& inputImg, Mat& outputImg);
+void filter2DFreq(const Mat& inputImg, Mat& outputImg, const Mat& H);
+void calcWnrFilter(const Mat& input_h_PSF, Mat& output_G, double nsr);
+
+int main(int argc, char *argv[])
+{
+    help();
+
+    int R = 53;
+    int snr = 5200;
+
+    string strInFileName = "original.JPG";
+
+    if (argc == 4)
+    {
+        strInFileName = argv[1];
+        R = atoi(argv[2]);
+        snr = atoi(argv[3]);
```

Comment:
OpenCV samples should use `cv::CommandLineParser`.

Reply:
I see